src/main/java/io/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -85,13 +85,38 @@ void update(T oldEntity, T entity) {
             currentSession().update(entity);
         }
 
+        void update(List<T> oldEntities, List<T> entities) {
+            for (T oldEntity : oldEntities) {
+                currentSession().evict(oldEntity);
+            }
+
+            if (entities != null) {
+                for (T entity : entities) {
+                    currentSession().update(entity);
+                }
+            }
+        }
+
+        void update(List<T> entities) {
+            if (entities != null) {
```

Comment:
return if it is null. Also the code is wrong i think as it does not call evict. It's way easier to call the update(old, new) for every element.